app/src/main/java/at/aau/edu/wizards/gameboard/GameboardAdapter.kt

```diff
@@ -0,0 +1,33 @@
+package at.aau.edu.wizards.gameboard
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import at.aau.edu.wizards.R
+import com.google.android.material.imageview.ShapeableImageView
+
+class GameboardAdapter(private val cardList: ArrayList<Cards>) : RecyclerView.Adapter<GameboardAdapter.MyViewHolder>() {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
+        val itemView = LayoutInflater.from(parent.context).inflate(
+            R.layout.card,
+            parent, false
+        )
+
+        return MyViewHolder(itemView)
+    }
+
+    override fun getItemCount(): Int {
+        return cardList.size
+    }
+
+    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
+        val currentItem = cardList[position]
+        holder.titleImage.setImageResource(currentItem.cardsImage)
+    }
+
+    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
```

Comment:
Should be called `CardViewHolder` or similar. I'd move it to separate file, but that's personal preference so up to you.